src/engine/strat_engine/devlinks.rs

```diff
@@ -3,169 +3,97 @@
-        let mut existing_dirs = fs::read_dir(DEV_PATH)?
-            .map(|dir_e| dir_e.map(|d| d.file_name().into_string().expect("Unix is utf-8")))
-            .collect::<Result<HashSet<_>, _>>()?;
-
-        for (pool_name, _, _) in pools {
-            existing_dirs.remove(&pool_name.to_owned());
-        }
-
-        for leftover in existing_dirs {
-            pool_removed(&Name::new(leftover));
+pub fn pool_renamed(pool_name: &str, _: &str) {
+    fn trigger_udev(pool_name: &str) -> StratisResult<()> {
+        let pool_dir: PathBuf = [DEV_PATH, pool_name].iter().collect();
+
+        for file_result in fs::read_dir(&pool_dir)? {
+            let file = file_result?;
+            let file_path = file.path();
+            let file_name = file_path
+                .file_name()
+                .and_then(|s| s.to_str())
+                .ok_or_else(|| {
+                    StratisError::Engine(
+                        ErrorEnum::Invalid,
+                        format!(
+                            "Failure while attempting to generate uevents for all \
+                            of the devices contained in directory {} to make /dev/stratis \
+                            symlinks consistent with internal pool state.",
+                            pool_dir.display(),
+                        ),
+                    )
+                })?;
+            filesystem_renamed(pool_name, file_name, file_name);
         }
 
         Ok(())
-    }() {
-        warn!("cleanup_devlinks failed, reason {:?}", err);
-    }
-}
-
-/// Create a directory when a pool is added.
-pub fn pool_added(pool: &str) {
-    let p = pool_directory(pool);
-    if let Err(e) = fs::create_dir(&p) {
-        warn!("unable to create pool directory {:?}, reason {:?}", p, e);
     }
-}
 
-/// Remove the directory and its contents when the pool is removed.
-pub fn pool_removed(pool: &str) {
-    let p = pool_directory(pool);
-    if let Err(e) = fs::remove_dir_all(&p) {
-        warn!("unable to remove pool directory {:?}, reason {:?}", p, e);
-    }
-}
-
-/// Rename the directory to match the pool's new name.
-pub fn pool_renamed(old_name: &str, new_name: &str) {
-    let old = pool_directory(old_name);
-    let new = pool_directory(new_name);
-    if let Err(e) = fs::rename(&old, &new) {
+    if let Err(e) = trigger_udev(pool_name) {
         warn!(
-            "unable to rename pool directory old {:?}, new {:?}, reason {:?}",
-            old, new, e
+            "Synthetic udev events were not able to be triggered: {}. Migration of \
+            filesystem links associated with renamed pool {} failed",
+            e, pool_name,
         );
     }
 }
 
-/// Create a symlink to the new filesystem's block device within its pool's
-/// directory.
-pub fn filesystem_added(pool_name: &str, fs_name: &str, devnode: &Path) {
-    let p = filesystem_mount_path(pool_name, fs_name);
+pub fn filesystem_renamed(pool_name: &str, fs_name: &str, _: &str) {
```

Comment:
Please put in a header comment, and also for method above.

Reply:
Resolved.